echopype/consolidate/loc_utils.py

```diff
@@ -0,0 +1,173 @@
+import numpy as np
+import xarray as xr
+
+from ..echodata import EchoData
+from ..utils.log import _init_logger
+
+logger = _init_logger(__name__)
+
+
+def check_loc_vars_present_all_NaN(
+    echodata: EchoData, lat_name: str, lon_name: str, datagram_type: str
+):
+    """
+    Checks if any of the location variables missing or are all NaN. Recommends using another
+    datagram_type if the corresponding lat/lon variables are not missing and not not all
+    NaN.
+    """
+    # Grab variables from EchoData object
+    lat_var = echodata["Platform"][lat_name]
+    lon_var = echodata["Platform"][lon_name]
+
+    # Check if loc vars missing or all NaN
+    any_var_missing = (lat_var is None) or (lon_var is None)
+    any_var_all_nan = lat_var.isnull().all() or lon_var.isnull().all()
+
+    if any_var_missing or any_var_all_nan:
+        # Initialize error message
+        error_message = "Coordinate variables not present or all nan."
+        if echodata.sonar_model.startswith("EK"):
+            # Initialize good datagram type list
+            good_datagram_types = []
+
+            # Iterate through potential datagram type sets
+            potential_datagram_type_set = set(["NMEA", "MRU1", "IDX"])
+            potential_datagram_type_set.discard(datagram_type)
+            for potential_datagram_type in potential_datagram_type_set:
+                # Grab variables from EchoData object
+                if potential_datagram_type != "NMEA":
+                    potential_lat_var = echodata["Platform"].get(
+                        f"latitude_{potential_datagram_type.lower()}"
+                    )
+                    potential_lon_var = echodata["Platform"].get(
+                        f"latitude_{potential_datagram_type.lower()}"
+                    )
+                elif potential_datagram_type == "NMEA":
+                    potential_lat_var = echodata["Platform"]["latitude"]
+                    potential_lon_var = echodata["Platform"]["longitude"]
+
+                # Check if all potential variables are present and not all NaN
+                all_present_and_not_all_nan = (
+                    potential_lat_var is not None
+                    and potential_lon_var is not None
+                    and not (potential_lat_var.isnull().all() or potential_lon_var.isnull().all())
+                )
+
+                if all_present_and_not_all_nan:
+                    good_datagram_types.append(potential_datagram_type)
+
+            if len(good_datagram_types) > 0:
+                # Append to error message
+                error_message = (
+                    error_message
+                    + f" Consider setting datagram_type to any of {good_datagram_types}."
+                )
+
+        # Raise error
+        raise ValueError(error_message)
+
+
+def check_loc_vars_any_NaN_0(echodata: EchoData, lat_name: str, lon_name: str, datagram_type: str):
```

Comment:
I feel that you can combine this and the above function (`check_loc_vars_present_all_NaN`). They are largely identical except for the actual checks.

Reply:
I agree, I'll make that change 👍